tests/components/tesla_fleet/test_sensor.py

```diff
@@ -41,3 +42,28 @@ async def test_sensors(
     await hass.async_block_till_done()
 
     assert_entities_alt(hass, normal_config_entry.entry_id, entity_registry, snapshot)
+
+
+@pytest.mark.usefixtures("entity_registry_enabled_by_default")
+async def test_sensors_restore(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    normal_config_entry: MockConfigEntry,
+    entity_registry: er.EntityRegistry,
+    freezer: FrozenDateTimeFactory,
+    mock_vehicle_data: AsyncMock,
+) -> None:
+    """Tests that the sensor entities are correct."""
+
+    freezer.move_to("2024-01-01 00:00:00+00:00")
+
+    # Setup platform to get some history data
+    await setup_platform(hass, normal_config_entry, [Platform.SENSOR])
+
+    # Set the vehicle offline
+    mock_vehicle_data.side_effect = VehicleOffline
+
+    # Reload the platform and allow restore entities to happen
+    with patch("homeassistant.components.tesla_fleet.PLATFORMS", [Platform.SENSOR]):
+        assert await hass.config_entries.async_reload(normal_config_entry.entry_id)
+    assert_entities_alt(hass, normal_config_entry.entry_id, entity_registry, snapshot)
```

Comment:
We just removed the restore functionality right? I am not sure if testing restore works well with snapshots. Yes, it covers all the entities, but it really hides away what you are actually trying to test

Reply:
I removed restore from some entities, and added a test that it returns a value on others.

I thought a snapshot of the restored values (because the entry was reloaded) would be the most effective.

A future improvement would be to also break things that dont have restore functionality so its extra clear.

Reply:
I think pinpointing "some" of the entities that should not restore likewise restore is best in this case than a flooded snapshot. 
Maybe as pytest params with online and then when offline as state tests.